test/retry_test.go

```diff
@@ -107,9 +108,9 @@ func TestTaskRunRetry(t *testing.T) {
 	pods, err := c.KubeClient.Kube.CoreV1().Pods(namespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("Failed to list Pods: %v", err)
-	} else if len(pods.Items) != numRetries {
+	} else if len(pods.Items) != numRetries+1 {
 		// TODO: Make this an error.
-		t.Logf("BUG: Found %d Pods, want %d", len(pods.Items), numRetries)
+		t.Logf("BUG: Found %d Pods, want %d", len(pods.Items), numRetries+1)
```

Comment:
Is it possible to make this an error now and fail the test?

Reply:
Sorry, my bad. the PR is only for correcting the test case, the issue mentioned in the issue #1976 is still existed need further research.

I prefer to keep current behavior(not an error just log) until the real issue fixed.

Reply:
Ah I totally understand, thanks @vincent-pli !  Could you update the commit message as well to reflect this?  Something like `Correct the number of expected pods created during retries e2e test`